## 1. map the reads

``
awk '{print "sbatch 1_AlignMito.slurm "$0}' 1_AlignMito.param | bash
``

## 2. get the consensus 

``
awk '{print "sbatch 2_ExtractMito.slurm "$2}' 1_AlignMito.param | bash
``

## 3. rename each sequence in the header 

``
for File in 2_MitoConsensus/*.fasta; do
echo $File
Name=`basename -s "_FullMito.fasta" $File`
echo ">"${Name} > 2_MitoConsensus/${Name}_FullMitoRenamed.fasta
tail -n +2 $File >> 2_MitoConsensus/${Name}_FullMitoRenamed.fasta
done
``

## 4. Identify the first and last gene present in the coding region of each mito-consensus (using MITOS)

run the 3_Mitos_annotation.slurm script as:
``
    for i in $(cat 1_samples_list.param); do
    sbatch 3_Mitos_annotation.slurm ${i}
    done
``
where 1_samples_list.param is the list of samples.

## 5. Repetitive elements present in the consensus mitogenomes should be removed
You need to index the mitoconsensus fasta file using samtools as:
``
samtools faidx my_mitoconsensus_FullMitoRenamed.fasta
``
Then using the information (genes and repetitive elements coordinates) generated by Mitos, select only the coding regions with samtools as:
``
  samtools faidx my_mitoconsensus_FullMitoRenamed.fasta header:1-250
``
where :1-250 correspond to the coordiantes (start-end) of the region you want to select

## 6. Alignment of all mito-consensus is generated using MATTF

## 7. Build a phylogenetic tree using maximum likelihood, using RAxML
